core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -67,9 +77,15 @@
   private static final String SERVICE_CONFIG_NAME_PREFIX = "_grpc_config.";
   // From https://github.com/grpc/proposal/blob/master/A5-grpclb-in-dns.md
   private static final String GRPCLB_NAME_PREFIX = "_grpclb._tcp.";
+  // From https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md
+  private static final String SERVICE_CONFIG_PREFIX = "_grpc_config=";
+  private static final Set<String> SERVICE_CONFIG_CHOICE_KEYS =
+      Collections.unmodifiableSet(
+          new HashSet<String>(
+              Arrays.asList("clientLanguage", "percentage", "clientHostname", "serviceConfig")));
```

Comment:
Just a question: Why is it not `ImmutableSet`?

Reply:
Guava collections adds a large number of methods to our library, which Android users would not like.   Android users have a limit of 64K methods per java program, and Guava takes from 5-10K.   Proguard gets rid of most of them, but not the collections classes.

Reply:
Thanks!